### ISSUE TITLE
FL A11Y: Placing text in a reasonable order

### ISSUE THREAD

**Issue by GarthConboy** [w3c/epub-specs#1571]
Wendy, thanks for kicking off these thoughts with: https://github.com/w3c/epub-specs/wiki/Fixed-Layout-Accessibility

One big step forward for "normal" HTML FL content would be that absolutely placed text on FL pages should be placed in the desired reading order such that a screen reader could be expected not to mangle it.

It seems this could become a SHOULD in EPUB 3.3.

**Comment by mattgarrish** [comment]
FYI: http://idpf.org/epub/profiles/edu/10/#h.n5rszhw79ymj

**Comment by GarthConboy** [comment]
Nice -- that looks covered!   :-)

**Comment by iherman** [comment]
> FYI: http://idpf.org/epub/profiles/edu/10/#h.n5rszhw79ymj

Should that text be moved to either the core text or the A11y document?

**Comment by mattgarrish** [comment]
I'm not sure if it belongs in either. It's not necessary in the a11y document as there's nothing there that isn't already covered by WCAG. And it doesn't belong in the core spec as we moved away from replicating WCAG requirements a couple of revisions ago.

Unless there's something new that isn't covered by WCAG, aren't we just looking at explainers of common problems and how to solve them? Wouldn't this fit more as a best practices document? Sort of a run-down of the common success criteria that fail for fixed layouts and how to avoid them?

The really tough problems with fixed layouts are things like images or content split across two-page spreads, providing descriptions (especially when an image occupies the entire page), ensuring text wraps when zoomed, etc. The solutions to these are never going to fit in a specification as there will always be variability in which method best fits which kind of content.

**Comment by mattgarrish** [comment]
Another consideration is that the accessibility requirements aren't going to be machine testable in all cases. Epubcheck isn't going to be able to tell you if elements are in the wrong DOM order, for example, as only a human could detect incorrect sequencing. That's also part of why we don't have accessibility requirements in the core anymore.

**Comment by iherman** [comment]
> Unless there's something new that isn't covered by WCAG,

I am not familiar enough with WCAG: is there really something in WCAG that would cover this?

**Comment by mattgarrish** [comment]
> is there really something in WCAG that would cover this

[SC 1.3.2](https://www.w3.org/TR/WCAG21/#meaningful-sequence) covers meaningful sequence and it's a Level A criterion so one of the basic requirements.

What I'm envisioning, at least, is a second techniques document just for FXL where we can highlight these specifics. EPUB sometimes puts a unique twist on how to meet WCAG, but WCAG generally covers everything we're going to want of publications whether reflowable or FXL.

**Comment by gregoriopellegrino** [comment]
Yes, in any case I think it would be useful to have best practices on how to make FXL EPUBs with some practical examples that put in relation WCAGs and use cases. (Something similar to DAISY KB, or maybe a new section there).

**Comment by avneeshsingh** [comment]
+1 to Matt and Gregorio.
WCAG principles are applicable to reflowable as well as fixed layout, in fact PDF can be made to conform to WCAG.
I also think that our focus should be on practical best practices, and may be hacks with which EPUB fixed layouts can be made accessible.

**Comment by mattgarrish** [comment]
> Something similar to DAISY KB, or maybe a new section there

There is a [section on FXL](https://kb.daisy.org/publishing/docs/fxl/index.html) in the KB, but I fully admit it's far from the most helpful in terms of examples (I just don't cross paths with a lot of real-world FXL).

I did do a run-through of WCAG some time back to pick out [the most problematic success criteria for FXL](https://kb.daisy.org/publishing/docs/fxl/wcag.html). Maybe that could help here as a starting point.
